protos/helpers/client.go

```diff
@@ -0,0 +1,128 @@
+package helpers
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+const (
+	defaultSchema = "http"
+)
+
+var (
+	domainRegexp = regexp.MustCompile(`^([a-zA-Z0-9-]{1,63}\.)*([a-zA-Z]{1,63})$`)
+	ipv4Regexp   = regexp.MustCompile(`^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$`)
+)
+
+type parsedURL struct {
+	schema string
+	host   string
+	port   int
+}
+
+func NewGRPCConn(ctx context.Context, address string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	u, err := parse(address)
+	if err != nil {
+		return nil, err
+	}
+	if u.schema == "http" {
+		opts = append(opts, grpc.WithInsecure())
+	} else if u.schema == "https" {
+		c := credentials.NewTLS(&tls.Config{})
+		opts = append(opts, grpc.WithTransportCredentials(c))
+	} else {
+		return nil, errors.New("not supported schema")
+	}
+	return grpc.DialContext(ctx, fmt.Sprintf("%s:%d", u.host, u.port), opts...)
+}
+
+func parse(rawU string) (*parsedURL, error) {
+	if strings.Index(rawU, "//") == 0 {
+		rawU = defaultSchema + ":" + rawU
+	}
+	if !strings.Contains(rawU, "://") {
+		rawU = defaultSchema + "://" + rawU
+	}
+	u, err := url.Parse(rawU)
+	if err != nil {
+		return nil, err
+	}
+	h, p, err := splitHostPort(u)
```

Comment:
Just found https://golang.org/pkg/net/url/#URL.Port, i think you can safely use url functions, which is more stable.

Reply:
will do it.

Reply:
url library doesn’t handle schema and port == "", and doesn’t check validaity of host. so i add some logic to do them.